pubsub/subscription.go

```diff
@@ -50,6 +51,11 @@ type Subscription struct {
 
 	mu            sync.Mutex
 	receiveActive bool
+
+	// clientID to be used across all streaming pull connections that are created.
+	// This indicates to the server that any guarantees made for a stream that
+	// disconnected will be made for the stream that is created to replace it.
+	clientID string
```

Comment:
are there use cases for making this visible to users?

Reply:
In Java the field is marked private, so until that changes, I'm incline to leave as-is.